mp_api/client/core/utils.py

```diff
@@ -12,6 +13,33 @@
 
 from mp_api.client.core.settings import MAPIClientSettings
 
+_MAPI_SETTINGS = MAPIClientSettings()
+
+
+def validate_api_key(api_key: str | None = None) -> str:
+    """Utility to find and pre-check validity of an API key."""
+    # SETTINGS tries to read API key from ~/.config/.pmgrc.yaml
+    api_key = api_key or os.getenv("MP_API_KEY")
+    if not api_key:
+        from pymatgen.core import SETTINGS
+
+        api_key = SETTINGS.get("PMG_MAPI_KEY")
+
+    if not api_key:
+        raise ValueError(
+            "Please obtain an API key from https://materialsproject.org/api "
+            "and export it as an environment variable `MP_API_KEY`."
+        )
+
+    if api_key and len(api_key) != 32:
+        raise ValueError(
+            "Please use a new API key from https://materialsproject.org/api "
+            "Keys for the new API are 32 characters, whereas keys for the legacy "
+            "API are 16 characters."
```

Comment:
I think we can start removing references to legacy API keys. The legacy services will go down in about a month